### PR TITLE
Add tests for `find_tasks_in_param`

### PR DESCRIPTION
**Pull request by nathanjmcdougall**
Add unit tests.

Some refactoring to have a one-source-of-truth for information about what scalars are supported in labtech tasks parameters.

### CODE REVIEW

**Comment by nathanjmcdougall** [comment]
Having type check issues, will take a look later

**Comment by nathanjmcdougall** [comment]
Okay, so tests are failing in 3.10 for the case of a frozendict; that `frozendict` is not actually a subtype of `dict` until 3.11. I'm guessing we should just explicitly handle this case in the function, but I want to check you you about that.

File: labtech/tasks.py [hunk 1]
```diff
@@ -255,4 +251,9 @@ def find_tasks_in_param(param_value: Any, searched_coll_ids: Optional[Set[int]]
             for item in param_value.values()
             for task in find_tasks_in_param(item, searched_coll_ids)
         ]
-    return []
+    elif isinstance(param_value, cast(UnionType, ParamScalar)):
+        return []
+
+    # This should be impossible.
+    msg = f"Unexpected type {type(param_value).__qualname__} encountered in task parameter value."
+    raise AssertionError(msg)
```
Comment (ben-denham) [inline labtech/tasks.py:259]: I believe `AssertionError` is only intended for use by `assert` statements, so I suggest we make this a `TaskError` for consistency with other errors in this file.

```suggestion
    raise TaskError(msg)
```

**Review by ben-denham — COMMENTED** [review]
Thanks again for the additional tests!

Yes, I agree it looks like I missed the handling of `frozendict` in `find_tasks_in_param()` that had been included in `immutable_param_value`, so `elif isinstance(param_value, dict):` should actually be `elif isinstance(param_value, dict) or isinstance(param_value, frozendict) :`

**Comment by ben-denham** [comment]
This branch is all good now, but needs a rebase on `main` to merge cleanly. I made the mistake of using GitHub's conflict resolution interface, but that's made an ugly merge commit on your branch sorry :(

**Review by ben-denham — APPROVED** [review]

**Pull request merged**